ipmininet/tests/test_ospf.py

```diff
@@ -0,0 +1,99 @@
+"""This module tests the OSPF daemon"""
+
+import pytest
+
+from ipmininet.clean import cleanup
+from ipmininet.examples.simple_ospf_network import SimpleOSPFNet
+from ipmininet.ipnet import IPNet
+from ipmininet.iptopo import IPTopo
+from ipmininet.router.config import OSPF
+from ipmininet.router.config.base import RouterConfig
+from ipmininet.router.config.ospf import OSPFRedistributedRoute
+from ipmininet.tests.utils import assert_connectivity
+from . import require_root
+
+
+class MinimalOSPFNet(IPTopo):
+    """
+
+    h1 ---- r1 ---- r2 ---- h2
+
+    """
+    def __init__(self, ospf_params_r1, link_params, *args, **kwargs):
+        """:param ospf_params_r1: Parameters to set on the OSPF daemon of r1
+        :param link_params: Parameters to set on the link between r1 and r2"""
+        self.ospf_params_r1 = ospf_params_r1
+        self.link_params = link_params
+        self.link_params.setdefault("params1", {}).setdefault("ip", "10.0.0.1/24")
+        self.link_params.setdefault("params2", {}).setdefault("ip", "10.0.0.2/24")
+        super(MinimalOSPFNet, self).__init__(*args, **kwargs)
+
+    def build(self, *args, **kwargs):
+        r1 = self.addRouter("r1", config=RouterConfig)
+        r1.addDaemon(OSPF, **self.ospf_params_r1)
+        r2 = self.addRouter("r2", config=RouterConfig)
+        r2.addDaemon(OSPF)
+        self.addLink(r1, r2, **self.link_params)
+
+        h1 = self.addHost("h1")
+        self.addLink(r1, h1, params1={"ip": "10.0.1.1/24"}, params2={"ip": "10.0.1.2/24"})
+        h2 = self.addHost("h2")
+        self.addLink(r2, h2, params1={"ip": "10.0.2.1/24"}, params2={"ip": "10.0.2.2/24"})
+        super(MinimalOSPFNet, self).build(*args, **kwargs)
+
+
+@require_root
+def test_ospf_example():
+    try:
+        net = IPNet(topo=SimpleOSPFNet())
+        net.start()
+        assert_connectivity(net)
+        net.stop()
+    finally:
+        cleanup()
+
+
+@require_root
+@pytest.mark.parametrize("ospf_params,link_params,expected_cfg", [
+    ({},
+     {},
+     ["  network 10.0.0.1/24 area 0.0.0.0",
+      "interface r1-eth0"]),
+    ({"debug": ["lsa"]},
+     {},
+     ["debug ospf lsa"]),
+    ({},
+     {"igp_metric": 2},
+     ["  ip ospf cost 2"]),
+    ({},
+     {"igp_area": "1.1.1.1"},
+     ["  network 10.0.0.1/24 area 1.1.1.1"]),
+    ({},
+     {"params1": {"ospf_priority": 1}},
+     ["  ip ospf priority 1"]),
+    ({},
+     {"params1": {"ospf_dead_int": "minimal hello-multiplier 2"}},
+     ["  ip ospf dead-interval minimal hello-multiplier 2"]),
+    ({"redistribute": [OSPFRedistributedRoute("connected", 1, 15), OSPFRedistributedRoute("static", 2, 50)]},
+     {},
+     ["  redistribute connected metric-type 1 metric 15",
+      "  redistribute static metric-type 2 metric 50"]),
```

Comment:
This is fine from a config generation point of view, but I'd add a parameter to those tests to describe the expected resulting paths.